audmath/core/api.py

```diff
@@ -86,6 +88,207 @@ def db(
+
+    Examples:
+        >>> duration_in_seconds(2)
+        2.0
+        >>> duration_in_seconds(2.0)
+        2.0
+        >>> duration_in_seconds('2ms')
+        0.002
+        >>> duration_in_seconds('2 ms')
+        0.002
+        >>> duration_in_seconds('ms')
+        0.001
+        >>> duration_in_seconds('2000', 1000)
+        2.0
+        >>> duration_in_seconds(np.timedelta64(2, 's'))
+        2.0
+        >>> duration_in_seconds(pd.to_timedelta(2, 's'))
+        2.0
+
+    """  # noqa: E501
+    # Dictionary with allowed unit entries
+    # and mapping from pandas.to_timedelta()
+    # to numpy.timedelta64, see
+    # https://numpy.org/doc/stable/reference/arrays.datetime.html#datetime-units
+    unit_mapping = {
+        # week
+        'W': 'W',
+        # day
+        'D': 'D',
+        'days': 'D',
+        'day': 'D',
+        # hour
+        'h': 'h',
+        'hours': 'h',
+        'hour': 'h',
+        'hr': 'h',
+        # minute
+        'm': 'm',
+        'minutes': 'm',
+        'minute': 'm',
+        'min': 'm',
+        'T': 'm',
+        # second
+        's': 's',
+        'seconds': 's',
+        'second': 's',
+        'sec': 's',
+        'S': 's',
+        # millisecond
+        'ms': 'ms',
+        'milliseconds': 'ms',
+        'millisecond': 'ms',
+        'millis': 'ms',
+        'milli': 'ms',
+        'L': 'ms',
+        # microsecond
+        'us': 'us',
+        'μs': 'us',
+        'microseconds': 'us',
+        'microsecond': 'us',
+        'micros': 'us',
+        'micro': 'us',
+        'U': 'us',
+        # nanosecond
+        'ns': 'ns',
+        'nanoseconds': 'ns',
+        'nanosecond': 'ns',
+        'nanos': 'ns',
+        'nano': 'ns',
+        'N': 'ns',
+    }
+
+    # numpy.timedelta64() accepts only integer as input values,
+    # so we need to convert all values
+    # to nanoseconds and integers first
+    def to_nanos(value, unit):
+        if unit == 'W':
+            value = value * 7 * 24 * 60 * 60 * 10 ** 9
+        elif unit == 'D':
+            value = value * 24 * 60 * 60 * 10 ** 9
+        elif unit == 'h':
+            value = value * 60 * 60 * 10 ** 9
+        elif unit == 'm':
+            value = value * 60 * 10 ** 9
+        elif unit == 's':
+            value = value * 10 ** 9
+        elif unit == 'ms':
+            value = value * 10 ** 6
+        elif unit == 'us':
+            value = value * 10 ** 3
+        return int(value)
+
+    if isinstance(duration, str):
+
+        # ensure we have a str and not numpy.str_
+        duration = str(duration).strip()
+
+        match = re.match(VALUE_UNBIT_PATTERN, duration)
+        if match is not None:
+            value, unit = match.groups()
+        if (
+                match is None
+                or (not value and not unit)
+        ):
+            raise ValueError(
+                f"Your given duration '{duration}' "
+                "is not conform to the 'value' 'unit' pattern."
```

Comment:
```suggestion
                "is not conform to the '<value><unit>' pattern."
```

Reply:
I changed the error message and the tests acordingly.